js/private/js_library.bzl

```diff
@@ -49,18 +49,27 @@ _ATTRS = {
         This includes all your checked-in code and any generated source files.
 
         The transitive npm dependencies, transitive sources & runfiles of targets in the `srcs` attribute are added to the
-        runfiles of this taregt. They should appear in the '*.runfiles' area of any executable which is output by or has a
+        runfiles of this target. They should appear in the '*.runfiles' area of any executable which is output by or has a
         runtime dependency on this target.
+
+        Source files that are JSON files, declaration files or directory artifacts will be automatically provided as
+        "declarations" available to downstream rules for type checking. To explicitly provide source files as "declarations"
+        available to downstream rules for type checking that do not match these criteria, use the `declarations`
+        attribute instead.
         """,
         allow_files = True,
     ),
+    "declarations": attr.label_list(
+        doc = """Same as `srcs` except all files are provided as "declarations" available to downstream rules for type checking.""",
```

Comment:
are also provided

Reply:
and maybe give an example of why you might need that.

Reply:
✅